napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -109,37 +117,28 @@ def __init__(self, viewer):
         self.rollDimsButton = QtViewerPushButton(
             self.viewer,
             'roll',
-            trans._("Roll dimensions order for display ({key}-E)").format(
-                key=KEY_SYMBOLS['Control']
-            ),
-            lambda: self.viewer.dims._roll(),
         )
         self.transposeDimsButton = QtViewerPushButton(
             self.viewer,
             'transpose',
-            trans._("Transpose displayed dimensions ({key}-T)").format(
-                key=KEY_SYMBOLS['Control']
-            ),
-            lambda: self.viewer.dims._transpose(),
-        )
-        self.resetViewButton = QtViewerPushButton(
-            self.viewer,
-            'home',
-            trans._("Reset view ({key}-R)").format(key=KEY_SYMBOLS['Control']),
-            lambda: self.viewer.reset_view(),
         )
 
+        action_manager.context['viewer'] = viewer
+
+        action_manager.bind_button('transpose_axes', self.transposeDimsButton)
+        action_manager.bind_button('roll_axes', self.rollDimsButton)
+
+        self.resetViewButton = QtViewerPushButton(self.viewer, 'home')
+
+        action_manager.bind_button('reset_view', self.resetViewButton)
+
         self.gridViewButton = QtStateButton(
             'grid_view_button',
             self.viewer.grid,
             'enabled',
             self.viewer.grid.events,
         )
-        self.gridViewButton.setToolTip(
-            trans._("Toggle grid view ({key}-G)").format(
-                key=KEY_SYMBOLS['Control']
-            )
-        )
+        action_manager.bind_button('toggle_grid', self.gridViewButton)
```

Comment:
I have been wondering, could we use a `QToolBar` instead and add QActions to it from the `action_manager` ?

We can explore this in a subsequent PR, but if we did this, we could get away with all the Manual Button creation we have and also the `bind_button` method, since adding a QAction to a QToolBar will create the button automagically.

We would need to be able to generate a QAction from the action_manager, and we could also remove the `bind_qaction` method.

Have something like:

```python
toolbar = QToolBar
toolbar.addAction(action_manager.qaction('toggle_grid'))
toolbar.addAction(action_manager.qaction('toggle_ndisplay'))
```

Reply:
Agreed – though I'm not familiar with QToolbar() but I didn't want to start this refactor.

IIRC on mac os they also support the user customisation.

Initially I have icons part of the action_manager (actually pairs of icons), but removed them to simplify.

Reply:
Too much for this PR indeed, but let's keep that in mind :)